### PR TITLE
Sync `upstream/master`

### PR DESCRIPTION
**Pull request by HalidOdat**

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+edition = "2018"
+publish = false
+
+[package.metadata]
+cargo-fuzz = true
+
+[dependencies]
+arbitrary = { version = "1", features = ["derive"] }
+libfuzzer-sys = "0.4"
+ryu = { path = ".." }
```
Comment (Razican) [inline fuzz/Cargo.toml]: Should this be `ryu_js`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,12 +43,10 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@nightly
-        with:
-          components: miri
+      - uses: dtolnay/rust-toolchain@miri
       - run: cargo miri test
         env:
-          MIRIFLAGS: "-Zmiri-tag-raw-pointers"
+          MIRIFLAGS: -Zmiri-strict-provenance
```
Comment (Razican) [inline .github/workflows/ci.yml:49]: I like that they are using miri here, maybe we could also use it in Boa.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,6 +5,12 @@ on:
   pull_request:
   schedule: [cron: "40 1 * * *"]
 
+permissions:
+  contents: read
+
+env:
+  RUSTFLAGS: -Dwarnings
```
Comment (Razican) [inline .github/workflows/ci.yml:12]: This could maybe be added to Boa too, to avoid having warnings in the repo.

File: fuzz/Cargo.toml (hunk 1 above)
Comment (HalidOdat) [reply fuzz/Cargo.toml]: Nice catch!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (HalidOdat) [reply .github/workflows/ci.yml:49]: Yes, I think so too maybe it could catch some hard to find bugs

**Review by Razican — APPROVED** [review]

**Review by RageKnify — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jedel1043) [reply .github/workflows/ci.yml:49]: Let's open an issue about this

**Pull request merged**